web-api/src/storage/postgres.rs

```diff
@@ -1263,3 +1272,60 @@ impl storage::Size for Storage {
         Ok(size)
     }
 }
+
+#[async_trait(?Send)]
+impl storage::IndexedProperties for Storage {
+    async fn load_schema(&self) -> Result<IndexedPropertiesSchema, Error> {
+        let mut tx = self.postgres.begin().await?;
+        let schema = self.load_schema_with_transaction(&mut tx).await?;
+        tx.commit().await?;
+        Ok(schema)
+    }
+
+    async fn extend_schema(
+        &self,
+        update: IndexedPropertiesSchemaUpdate,
+        max_properties: usize,
+    ) -> Result<IndexedPropertiesSchema, Error> {
+        let mut tx = self.postgres.begin().await?;
+        let mut schema = self.load_schema_with_transaction(&mut tx).await?;
+        schema.update(&update, max_properties)?;
+        self.extend_postgres_schema(&mut tx, &update).await?;
+        //TODO self.elastic.extend_elasticsearch_mapping(&update).await?;
+        tx.commit().await?;
+        Ok(schema)
+    }
+}
+
+impl Storage {
```

Comment:
🔧 it looks like these impls should be implemented on `Database` instead, like the other postgres specific impls above.